hudi-aws/src/test/java/org/apache/hudi/aws/sync/TestAWSGlueSyncClient.java

```diff
@@ -145,4 +152,93 @@ void testDropTable() {
     // verify if aws glue delete table method called once
     verify(mockAwsGlue, times(1)).deleteTable(any(DeleteTableRequest.class));
   }
+
+  @Test
+  void testMetastoreFieldSchemas() {
+    String tableName = "testTable";
+    String databaseName = "testdb";
+    String inputFormatClass = "inputFormat";
+    String outputFormatClass = "outputFormat";
+    String serdeClass = "serde";
+    HashMap<String, String> serdeProperties = new HashMap<>();
+    HashMap<String, String> tableProperties = new HashMap<>();
+    List<Column> columns = Arrays.asList(Column.builder().name("name").type("string").comment("person's name").build(),
+        Column.builder().name("age").type("int").comment("person's age").build());
+    List<Column> partitionKeys = Arrays.asList(Column.builder().name("city").type("string").comment("person's city").build());
+    software.amazon.awssdk.services.glue.model.StorageDescriptor storageDescriptor = software.amazon.awssdk.services.glue.model.StorageDescriptor.builder()
+        .serdeInfo(SerDeInfo.builder().serializationLibrary(serdeClass).parameters(serdeProperties).build())
+        .inputFormat(inputFormatClass)
+        .columns(columns)
+        .outputFormat(outputFormatClass)
+        .build();
+    Table tempTable = Table.builder()
+        .name(tableName)
+        .tableType("COPY_ON_WRITE")
+        .parameters(new HashMap<>())
+        .storageDescriptor(storageDescriptor)
+        .partitionKeys(partitionKeys)
+        .parameters(tableProperties)
+        .databaseName(databaseName)
+        .build();
+    GetTableResponse response = GetTableResponse.builder()
+        .table(tempTable)
+        .build();
+
+    CompletableFuture<GetTableResponse> tableResponse = CompletableFuture.completedFuture(response);
+    // mock aws glue get table call
+    Mockito.when(mockAwsGlue.getTable(any(GetTableRequest.class))).thenReturn(tableResponse);
+
+    List<FieldSchema> fields = awsGlueSyncClient.getMetastoreFieldSchemas(tableName);
+    assertEquals(3, fields.size(), "Glue table schema contain 3 fields");
+    assertEquals("name", fields.get(0).getName(), "glue table first column should be name");
+    assertEquals("int", fields.get(1).getType(), "glue table second column type should be int");
+    assertEquals("person's city", fields.get(2).getComment().get(), "glue table third column comment should be age");
+  }
+
+  @Test
+  void testMetastoreFieldSchemas_EmptyPartitions() {
+    String tableName = "testTable";
```

Comment:
Please do not copy-paste. Extract common code of creating a table to private method and invoke it from both test-cases. DRY, you know.

Reply:
Fixed it.